File: backend/src/main/java/com/ll/TeamProject/domain/user/service/ForbiddenService.java
package com.ll.TeamProject.domain.user.service;

import com.ll.TeamProject.domain.user.dto.ForbiddenNicknameList;
import com.ll.TeamProject.domain.user.repository.ForbiddenRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ForbiddenService {
    private final ForbiddenRepository forbiddenRepository;

    public boolean isForbidden(String nickname) {
        ForbiddenNicknameList forbiddenList = new ForbiddenNicknameList(forbiddenRepository.findAll());

        return forbiddenList.contains(nickname);
    }
}

File: backend/src/main/java/com/ll/TeamProject/domain/user/service/ForbiddenService.kt
package com.ll.TeamProject.domain.user.service

import com.ll.TeamProject.domain.user.dto.ForbiddenNicknameList
import com.ll.TeamProject.domain.user.repository.ForbiddenRepository
import org.springframework.stereotype.Service

@Service
class ForbiddenService(
    private val forbiddenRepository: ForbiddenRepository
) {
    fun isForbidden(nickname: String): Boolean {
        val forbiddenList = ForbiddenNicknameList(forbiddenRepository.findAll())
        return forbiddenList.contains(nickname)
    }
}
